distributed/scheduler.py

```diff
@@ -4907,7 +4914,31 @@ async def update_graph(
                 start=start,
                 stimulus_id=stimulus_id or f"update-graph-{start}",
             )
-        except RuntimeError as e:
+            task_state_created = time()
+            metrics.update(
+                {
+                    "start": start,
+                    "duration_materialization": materialization_done - start,
+                    "duration_ordering": materialization_done - ordering_done,
+                    "duration_state_initialization": ordering_done - task_state_created,
+                    "duration_total": task_state_created - start,
+                }
+            )
+            evt_msg = {
+                "action": "update-graph",
+                "stimulus_id": stimulus_id,
+                "metrics": metrics,
+                "status": "OK",
+            }
+            self.log_event(["all", client, "update-graph"], evt_msg)
+            logger.debug("Task state created. %i new tasks", len(self.tasks) - before)
+        except Exception as e:
+            evt_msg = {
+                "action": "update-graph",
+                "stimulus_id": stimulus_id,
+                "status": "error",
+            }
+            self.log_event(["all", client, "update-graph"], evt_msg)
```

Comment:
No particular reason. I used this in a test but I will change it. I will also drop the `all` topic (feels a bit like an anti pattern)

Reply:
Note that dropping the `all` pattern is a user-facing breaking change. That being said, I'm all for redesigning our topics, etc., this might just require some changes for downstream users, e.g., for Coiled.

Reply:
Then again, we've also renamed the action, so these changes are breaking already.

Reply:
Yeah, I think this is breaking and I'm fine with it. This is overall a pretty burried feature and I doubt (m)any users will notice.